From a8b30d81a68d6edfde9769dcde19f22c7ddf428d Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:19 +0100
Subject: [PATCH 25/52] Revert "Add method to forget generated declarations"

This reverts commit c5272fe8b7125b8776a25f4fcf90874bd0f87b96.
---
 .../clang/include/clang/CodeGen/ModuleBuilder.h          | 1 -
 .../llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp     | 9 ---------
 2 files changed, 10 deletions(-)

diff --git a/interpreter/llvm-project/clang/include/clang/CodeGen/ModuleBuilder.h b/interpreter/llvm-project/clang/include/clang/CodeGen/ModuleBuilder.h
index f516309c16..b6c8a14bbf 100644
--- a/interpreter/llvm-project/clang/include/clang/CodeGen/ModuleBuilder.h
+++ b/interpreter/llvm-project/clang/include/clang/CodeGen/ModuleBuilder.h
@@ -106,7 +106,6 @@ public:
                             const CodeGenOptions& CGO);
 
   void forgetGlobal(llvm::GlobalValue* GV);
-  void forgetDecl(llvm::StringRef MangledName);
 };
 
 /// CreateLLVMCodeGen - Create a CodeGenerator instance.
diff --git a/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp b/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp
index f6549f070e..f4a5e0fc68 100644
--- a/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp
+++ b/interpreter/llvm-project/clang/lib/CodeGen/ModuleBuilder.cpp
@@ -283,11 +283,6 @@ namespace clang {
       }
     }
 
-    void forgetDecl(llvm::StringRef MangledName) {
-      Builder->DeferredDecls.erase(MangledName);
-      Builder->Manglings.erase(MangledName);
-    }
-
     void Initialize(ASTContext &Context) override {
       Ctx = &Context;
 
@@ -514,10 +509,6 @@ void CodeGenerator::forgetGlobal(llvm::GlobalValue* GV) {
   static_cast<CodeGeneratorImpl*>(this)->forgetGlobal(GV);
 }
 
-void CodeGenerator::forgetDecl(llvm::StringRef MangledName) {
-  static_cast<CodeGeneratorImpl*>(this)->forgetDecl(MangledName);
-}
-
 CodeGenerator *
 clang::CreateLLVMCodeGen(DiagnosticsEngine &Diags, llvm::StringRef ModuleName,
                          IntrusiveRefCntPtr<llvm::vfs::FileSystem> FS,
-- 
2.43.5

